app/src/main/java/app/grapheneos/camera/CamConfig.kt

```diff
@@ -1492,20 +1506,15 @@ class CamConfig(private val mActivity: MainActivity) {
                     val format = BarcodeFormat.valueOf(optionName)
 
                     if (optionValue) {
-                        allowedFormats.add(format)
-
-                        putBoolean(formatSRep, true)
-                    } else if (format in allowedFormats) {
-                        if (allowedFormats.size == 1) {
-                            mActivity.showMessage(
-                                getString(R.string.no_barcode_selected)
-                            )
-                        } else {
+                        if (format !in allowedFormats)
+                            allowedFormats.add(format)
+                    } else {
+                        if (format in allowedFormats) {
```

Comment:
`format in allowedFormats` check is redundant, `allowedFormats.remove(format)` handles this itself.